alg/teca_mesh_padding.cxx

```diff
@@ -0,0 +1,281 @@
+        return nullptr;
+    }
+
+    // create output and copy metadata, coordinates, etc
+    p_teca_cartesian_mesh out_mesh = teca_cartesian_mesh::New();
+
+    out_mesh->copy_metadata(
+        std::const_pointer_cast<teca_cartesian_mesh>(in_mesh));
+
+    teca_metadata &out_metadata = out_mesh->get_metadata();
+
+    size_t py_low = this->get_py_low();
+    size_t py_high = this->get_py_high();
+    size_t px_low = this->get_px_low();
+    size_t px_high = this->get_px_high();
+
+    if (py_low || py_high || px_low || px_high)
+    {
+        // get the coordinate axes
+        std::string y_name;
+        std::string x_name;
+        in_mesh->get_y_coordinate_variable(y_name);
+        in_mesh->get_x_coordinate_variable(x_name);
+
+        const_p_teca_variant_array y_crds = in_mesh->get_y_coordinates();
+        const_p_teca_variant_array x_crds = in_mesh->get_x_coordinates();
+        const_p_teca_variant_array z_crds = in_mesh->get_z_coordinates();
+
+        size_t n_y = y_crds->size();
+        size_t n_x = x_crds->size();
+        size_t n_z = z_crds->size();
+
+        size_t ny_new = py_low + n_y + py_high;
+        size_t nx_new = px_low + n_x + px_high;
+
+        size_t nxy_new = nx_new * ny_new;
+
+
+        unsigned long req_whole_extent[6] = {0};
+        unsigned long req_extent[6] = {0};
+
+        req_whole_extent[1] = nx_new - 1;
+        req_whole_extent[3] = ny_new - 1;
+        req_whole_extent[5] = n_z;
+        req_extent[1] = nx_new - 1;
+        req_extent[3] = ny_new - 1;
+        req_extent[5] = n_z;
+
+        out_mesh->set_whole_extent(req_whole_extent);
+        out_mesh->set_extent(req_extent);
+
+        out_metadata.set("whole_extent", req_whole_extent, 6);
+        out_metadata.set("extent", req_extent, 6);
+
+
+        p_teca_variant_array x_crds_new = x_crds->new_instance(px_low + n_x + px_high);
+
+        TEMPLATE_DISPATCH_FP(
+            teca_variant_array_impl,
+            x_crds_new.get(),
+
+            const NT* p_x_crds = static_cast<const TT*>(x_crds.get())->get();
+            NT* p_x_crds_new = static_cast<TT*>(x_crds_new.get())->get();
+
+            teca_algorithm_util::expand_dimension(p_x_crds_new, p_x_crds, n_x,
+                            px_low, px_high);
+
+            out_mesh->set_x_coordinates(const_cast<const std::string&>(x_name), x_crds_new);
+        )
+
+        p_teca_variant_array y_crds_new = y_crds->new_instance(py_low + n_y + py_high);
+
+        TEMPLATE_DISPATCH_FP(
+            teca_variant_array_impl,
+            y_crds_new.get(),
+
+            const NT* p_y_crds = static_cast<const TT*>(y_crds.get())->get();
+            NT* p_y_crds_new = static_cast<TT*>(y_crds_new.get())->get();
+
+            teca_algorithm_util::expand_dimension(p_y_crds_new, p_y_crds, n_y,
+                            py_low, py_high);
+
+            out_mesh->set_y_coordinates(const_cast<const std::string&>(y_name), y_crds_new);
+        )
+
```

Comment:
handling 3d (z dimension) is missing.Similarly, if given 1d data this would crash.  I guess you don't need these for your current set of inputs, but I think these could be handled with some trivial changes.  Also it is safe to assume that the coordinates all have the same type. Doing so would allow you to simplify and reduce this code.

Reply:
Do you mean wrap this repeated part of code into a function in the namespace above?

Reply:
this class will go into a separate pr